salt/modules/napalm_network.py

```diff
@@ -1946,55 +1950,66 @@ def load_template(template_name,
-                # no saltenv overridden
-                # use the custom template path
-                saltenv = 'base'
-            elif salt_render and not saltenv:
-                # if saltenv not overridden and path specified as salt:// or http:// etc.
-                # will use the default environment, from the base
-                saltenv = 'base'
-
-            if not saltenv:
-                # still not specified, default to `base`
-                saltenv = 'base'
             # render the file - either local, either remote
-            _rand_filename = __salt__['random.hash'](template_name, 'md5')
-            _temp_file = __salt__['file.join']('/tmp', _rand_filename)
-            _managed = __salt__['file.get_managed'](name=_temp_file,
-                                                    source=template_name,  # abs path
-                                                    source_hash=template_hash,
-                                                    source_hash_name=template_hash_name,
-                                                    user=template_user,
-                                                    group=template_group,
-                                                    mode=template_mode,
-                                                    attrs=template_attrs,
-                                                    template=template_engine,
-                                                    context=template_vars,
-                                                    defaults=defaults,
-                                                    saltenv=saltenv,
-                                                    skip_verify=skip_verify)
-            if not isinstance(_managed, (list, tuple)) and isinstance(_managed, six.string_types):
-                _loaded['comment'] = _managed
-                _loaded['result'] = False
-            elif isinstance(_managed, (list, tuple)) and not len(_managed) > 0:
-                _loaded['result'] = False
-                _loaded['comment'] = 'Error while rendering the template.'
-            elif isinstance(_managed, (list, tuple)) and not len(_managed[0]) > 0:
-                _loaded['result'] = False
-                _loaded['comment'] = _managed[-1]  # contains the error message
-            if _loaded['result']:  # all good
-                _temp_tpl_file = _managed[0]
-                _temp_tpl_file_exists = __salt__['file.file_exists'](_temp_tpl_file)
-                if not _temp_tpl_file_exists:
+            if not isinstance(template_name, (list, tuple)):
+                template_name = [template_name]
+            if template_hash_name and not isinstance(template_hash_name, (list, tuple)):
+                template_hash_name = [template_hash_name]
+            elif not template_hash_name:
+                template_hash_name = [None] * len(template_name)
+            if template_hash and isinstance(template_hash, six.string_types) and not\
```

Comment:
I was wondering what's @terminalmage's take on this? Does this make sense to you, or would you suggest doing this in a better way? Thanks.

Reply:
Once this is added, this should be good to merge IMO